test/FunctionalTestUtils20/InProcessServer.cs

```diff
@@ -3,34 +3,52 @@
     using System;
     using System.IO;
     using Microsoft.AspNetCore.Hosting;
-    using AI;
+    using Microsoft.Extensions.DependencyInjection;
+    using Xunit.Abstractions;
+
+    public class EndpointAddress
+    {
+        private const string httpListenerConnectionString = "http://localhost:{0}/v2/track/";
+        private static Random random = new Random();
+
+        public EndpointAddress()
+        {
+            this.ConnectionString = string.Format(httpListenerConnectionString, random.Next(1000, 5000).ToString());
+        }
+
+        public string ConnectionString { get; }
+    }
 
     // a variant of aspnet/Hosting/test/Microsoft.AspNetCore.Hosting.Tests/HostingEngineTests.cs
     public class InProcessServer : IDisposable
     {
-        private const string httpListenerConnectionString = "http://localhost:4001/v2/track/";
+        private readonly string httpListenerConnectionString;
+        private readonly ITestOutputHelper output;
 
         private static Random random = new Random();
         private static object noParallelism = new object();
 
-        public static Func<IWebHostBuilder, IWebHostBuilder> UseApplicationInsights =
-            builder => builder.UseApplicationInsights();
-
-        private readonly Func<IWebHostBuilder, IWebHostBuilder> configureHost;
         private IWebHost hostingEngine;
         private string url;
 
         private TelemetryHttpListenerObservable listener;       
         
 
-        public InProcessServer(string assemblyName, Func<IWebHostBuilder, IWebHostBuilder> configureHost = null)
+        public InProcessServer(string assemblyName, ITestOutputHelper output)
         {
-            this.configureHost = configureHost;
+            this.output = output;
 
             var machineName = Environment.GetEnvironmentVariable("COMPUTERNAME");
-            this.url = "http://" + machineName + ":" + random.Next(5000, 14000).ToString();           
+            this.url = "http://" + machineName + ":" + random.Next(5000, 14000).ToString();
+
+            output.WriteLine(string.Format("{0}: Launching application at: {1}", DateTime.Now.ToString("G"), this.url));
```

Comment:
can you log assembly name, and the name of the test itself? it'll help if we are looking at logs to investigate any failures.